arcjet/index.ts

```diff
@@ -524,6 +570,180 @@ export function slidingWindow<
+    return "CREDIT_CARD_NUMBER";
+  }
+
+  if (entity.tag === "phone-number") {
+    return "PHONE_NUMBER";
+  }
+
+  return entity.val;
+}
+
+function convertAnalyzeDetectedEntity(
+  detectedEntities: DetectedEntity[],
+): ArcjetIdentifiedEntity[] {
+  return detectedEntities.map((detectedEntity) => {
+    return {
+      ...detectedEntity,
+      identifiedType: analyzeEntitiesToString(detectedEntity.identifiedType),
+    };
+  });
+}
+
+export function sensitiveInfo<
+  const Detect extends DetectEntities<CustomEntities>,
+  const CustomEntities extends string,
+>(
+  options: SensitiveInfoOptions<Detect, CustomEntities>,
+  ...additionalOptions: SensitiveInfoOptions<Detect, CustomEntities>[]
+): Primitive<{}> {
+  const rules: ArcjetSensitiveInfoRule<{}>[] = [];
+
+  // Always create at least one SENSITIVE_INFO rule
+  for (const opt of [options, ...additionalOptions]) {
+    const mode = opt.mode === "LIVE" ? "LIVE" : "DRY_RUN";
+    if (typeof opt.allow !== "undefined" && typeof opt.deny !== "undefined") {
+      throw new Error(
+        "Both allow and deny cannot be provided to sensitiveInfo",
+      );
+    }
+
+    rules.push({
+      type: "SENSITIVE_INFO",
+      priority: Priority.SensitiveInfo,
+      mode,
+      allow: opt.allow || [],
+      deny: opt.deny || [],
+
+      validate(
+        context: ArcjetContext,
+        details: ArcjetRequestDetails,
+      ): asserts details is ArcjetRequestDetails {},
+
+      async protect(
+        context: ArcjetContext,
+        details: ArcjetRequestDetails,
+      ): Promise<ArcjetRuleResult> {
+        const body = await context.getBody();
+        if (typeof body === "undefined") {
+          return new ArcjetRuleResult({
+            ttl: 0,
+            state: "NOT_RUN",
+            conclusion: "ERROR",
+            reason: new ArcjetErrorReason(
+              "Couldn't read the body of the request to perform sensitive info identification.",
+            ),
+          });
+        }
+
+        let convertedDetect = undefined;
+        if (typeof opt.detect !== "undefined") {
+          const detect = opt.detect;
+          convertedDetect = (tokens: string[]) => {
+            return detect(tokens)
+              .filter((e) => typeof e !== "undefined")
+              .map(protocolEntitiesToAnalyze);
+          };
+        }
+
+        let entitiesTag: "allow" | "deny" = "allow";
+        let entitiesVal: Array<ReturnType<typeof protocolEntitiesToAnalyze>> =
+          [];
+
+        if (Array.isArray(opt.allow)) {
+          entitiesTag = "allow";
+          entitiesVal = opt.allow
+            .filter((e) => typeof e !== "undefined")
+            .map(protocolEntitiesToAnalyze);
+        }
+
+        if (Array.isArray(opt.deny)) {
+          entitiesTag = "deny";
+          entitiesVal = opt.deny
+            .filter((e) => typeof e !== "undefined")
+            .map(protocolEntitiesToAnalyze);
+        }
+
+        const entities = {
```

Comment:
I was looking at the naming around these WIT types and it is really generic. We need to fix them before we ship this as they overlap with so much else we want to define.